presto-hive/src/main/java/com/facebook/presto/hive/PathFilterProvider.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import org.apache.hadoop.fs.PathFilter;
+
+public interface PathFilterProvider
```

Comment:
Let's rename it to `FileFormatSpecificPathFilterFactory`, as it always creates a new one.

Reply:
Also please leave a comment on that interface explaining what it is for. 

It might not be trivial for somebody who is not in the loop to understand that.

Reply:
Done!